client/components/RequireAuth.tsx

```diff
@@ -2,18 +2,13 @@ import React, { PropsWithChildren } from 'react';
 import { Navigate, useLocation } from 'react-router-dom';
 import { useAuth } from '../auth/AuthProvider';
 
-interface HeaderProps {
-    children: any;
-}
-
-// function RequireAuth(): JSX.Element {
-// eslint-disable-next-line react/function-component-definition
-const RequireAuth: React.FC<PropsWithChildren<HeaderProps>> = ({ children }) => {
-    // <div>{children}</div>
-
-    const auth = useAuth();
+function RequireAuth({ children }: React.PropsWithChildren<unknown>) {
+    const { user, loading } = useAuth();
+    // if authentication is still being confirmed, return a loading component
+    // TODO: replace this with a loading component
+    if (loading) return <>Loading...</>;
```

Comment:
@milos381 a generic `Loading` component could be useful to display during authentication or when we're fetching data for the dashboard.

Reply:
I have now implemented a `Loading` component that just says "loading". It should be updated to a spinner or something.